controllers/pvc-mutator/storageclass/annotation_test.go

```diff
@@ -0,0 +1,148 @@
+package storageclass
+
+import (
+	"context"
+	"testing"
+
+	"github.com/storageos/api-manager/internal/pkg/provisioner"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	kscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestMutatePVC(t *testing.T) {
+	// Create a new scheme and add all the types from different clientsets.
+	scheme := runtime.NewScheme()
+	if err := kscheme.AddToScheme(scheme); err != nil {
+		t.Fatal(err)
+	}
+
+	// Default StorageOS StorageClass.
+	defStosSC := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:  types.UID("storageos-default-uid"),
+			Name: "stos-default",
+		},
+		Provisioner: provisioner.DriverName,
+	}
+
+	// StorageOS StorageClass.
+	stosSC := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:  types.UID("storageos-uid"),
+			Name: "stos",
+		},
+		Provisioner: provisioner.DriverName,
+	}
+
+	// Non-StorageOS StorageClass.
+	notStosSC := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:  types.UID("foo-uid"),
+			Name: "non-stos",
+		},
+		Provisioner: "foo-provisioner",
+	}
+
+	testNamespace := "default"
+
+	testcases := []struct {
+		name         string
+		namespace    string
+		storageClass *storagev1.StorageClass
+		wantErr      bool
+	}{
+		{
+			name:      "empty StorageClass",
+			namespace: testNamespace,
+			wantErr:   true,
```

Comment:
Empty StorageClass shouldn't cause the mutation to return error. `StorageClassForPVC()` will return the default cluster StorageClass value if it's not set in the PVC.
For a failure case, setting the PVC StorageClass to a non-existent SC will work.

Reply:
I removed this created a separate case for the failure.